openmp/tools/omptest/CMakeLists.txt

```diff
@@ -0,0 +1,137 @@
+
+# Target: ompTest library
+# On (implicit) request of GoogleTest, link against the one provided with LLVM.
+if ((NOT LIBOMPTEST_BUILD_STANDALONE) OR LIBOMPTEST_BUILD_UNITTESTS)
+  # Check if standalone build was requested together with unittests
+  if (LIBOMPTEST_BUILD_STANDALONE)
+    # Emit warning: this build actually depends on LLVM's GoogleTest
+    message(Warning "LIBOMPTEST_BUILD_STANDALONE and LIBOMPTEST_BUILD_UNITTESTS"
+                    " requested simultaneously.\n"
+                    "Linking against LLVM's GoogleTest library archives.\n"
+                    "Disable LIBOMPTEST_BUILD_UNITTESTS to perform an actual"
+                    " standalone build.")
+    # Explicitly disable LIBOMPTEST_BUILD_STANDALONE
+    set(LIBOMPTEST_BUILD_STANDALONE OFF)
+  endif()
+
+  # Use LLVM's gtest library archive
+  set(GTEST_LIB "${LLVM_BINARY_DIR}/lib/libllvm_gtest.a")
+
+  # Link gtest as whole-archive to expose required symbols
+  set(GTEST_LINK_CMD "-Wl,--whole-archive" ${GTEST_LIB}
+                     "-Wl,--no-whole-archive" LLVMSupport)
+
+  # Add GoogleTest-based header
+  target_sources(omptest PRIVATE ./include/OmptTesterGoogleTest.h)
+
+  # Add LLVM-provided GoogleTest include directories.
+  target_include_directories(omptest PRIVATE
+    ${LLVM_THIRD_PARTY_DIR}/unittest/googletest/include)
+
+  # TODO: Re-visit ABI breaking checks, disable for now.
+  target_compile_definitions(omptest PUBLIC
+    -DLLVM_DISABLE_ABI_BREAKING_CHECKS_ENFORCING)
+
+  # Link against gtest and gtest_main
+  target_link_libraries(omptest PRIVATE ${GTEST_LINK_CMD})
+else()
+  # Add 'standalone' compile definitions
+  target_compile_definitions(omptest PRIVATE
+    -DOPENMP_LIBOMPTEST_BUILD_STANDALONE)
+
+  # Add 'standalone' source files
+  target_sources(omptest PRIVATE
+    ./include/OmptTesterStandalone.h
+    ./src/OmptTesterStandalone.cpp)
+endif()
+
+# Add common include directories.
+target_include_directories(omptest PUBLIC
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+  $<INSTALL_INTERFACE:$<INSTALL_PREFIX>/${LIBOMP_HEADERS_INSTALL_PATH}/omptest>
+)
+
+target_compile_features(omptest PRIVATE cxx_std_17)
+
+# Create and install package configuration files.
+configure_package_config_file(
+  ${CMAKE_CURRENT_SOURCE_DIR}/cmake/omptest-config.cmake.in
+  ${CMAKE_CURRENT_BINARY_DIR}/cmake/omptest-config.cmake
+  INSTALL_DESTINATION "${OPENMP_INSTALL_LIBDIR}/cmake/openmp/omptest"
+)
+
+install(FILES ${omptest_BINARY_DIR}/cmake/omptest-config.cmake
+        DESTINATION "${OPENMP_INSTALL_LIBDIR}/cmake/openmp/omptest")
+
+# Install libomptest header files: Copy header-files from include dir
+install(DIRECTORY ./include/
+        DESTINATION "${LIBOMP_HEADERS_INSTALL_PATH}/omptest"
+        FILES_MATCHING PATTERN "*.h")
+
+# Install library and export targets.
+# Note: find_package(omptest) may require setting of PATH_SUFFIXES
+#       Example: "lib/cmake/openmp/omptest", this is due to the install location
+install(TARGETS omptest
```

Comment:
Why is this being installed? If I understand correctly, this is a testing library for implementors (i.e. for use by LLVM developers) not for openmp users, so I would not expect it to get installed, just like other testing libraries and binaries.

Reply:
Yes, agreed, I'll guard it so it is only installed when people actively request it.

Reply:
Thanks for pointing this out, PR is up: https://github.com/llvm/llvm-project/pull/155433